doc/install.rst

```diff
@@ -1,58 +1,655 @@
+then build Arbor is:
+
+.. code-block:: bash
+
+    export CRAYPE_LINK_TYPE=dynamic
+    module swap PrgEnv-cray PrgEnv-gnu
+    moudle swap gcc/7.1.0
+    export CC=`which cc`; export CXX=`which CC`;
+    cmake .. -DARB_DISTRIBUTED_MODEL=mpi \      # MPI support
+             -DCMAKE_BUILD_TYPE=release  \      # optimized
+             -DARB_THREADING_MODEL=tbb   \      # tbb threading
+             -DARB_SYSTEM_TYPE=Cray             # turn on Cray specific options
+
+.. Note::
+    If ``CRAYPE_LINK_TYPE`` isn't set, there will be warnings like the following when linking:
+
+    .. code-block:: none
+
+        warning: Using 'dlopen' in statically linked applications requires at runtime
+                 the shared libraries from the glibc version used for linking
+
+    Often the library or executable will work, however if a different glibc is loaded,
+    Arbor will crash at runtime with obscure errors that are very difficult to debug.
+
+
+.. _troubleshooting:
+
+Troubleshooting
+===============
+
+.. _crosscompiling:
+
+Cross Compiling NMODL
+---------------------
+
+Care must be taken when Arbor is compiled on a system with a different architecture to the target system where Arbor will run.
+This occurs quite frequently on HPC systems, for example when building on a login/service node that has a different architecture to the compute nodes.
+
+.. Note::
+    If building Arbor on a laptop or desktop system, i.e. on the same computer that
+    you will run Arbor on, cross compilation is not an issue.
+
+.. Warning::
+    ``Illegal instruction`` errors are a sure sign that
+    Arbor is running on a system that does not support the architecture it was compiled for.
+
+When cross compiling, we have to take care that the *modcc* compiler, which is used to convert NMODL to C++/CUDA code, is able to run on the compilation node.
+
+By default, CMake looks for the *modcc* executable, ``modcc``, in paths specified by the ``PATH`` environment variable, and will use this executable if it finds it.
+Otherwise, the CMake script will build *modcc* from source.
+To ensure that cross compilation works, a copy of modcc that is compiled for the build system should be in ``PATH``.
+
+Here we will use the example of compiling for Intel KNL on a Cray system, which has Intel Sandy Bridge CPUs on login nodes that don't support the AVX512 instructions used by KNL.
+
+
+.. code-block:: bash
+
+    #
+    #   Step 1: Build modcc.
+    #
+
+    module swap PrgEnv-cray PrgEnv-gnu
+    # Important: use GNU compilers directly, not the compiler wrappers,
+    # which generate code for KNL, not the login nodes.
+    export CC=`which gcc`; export CXX=`which g++`;
+    export CRAYPE_LINK_TYPE=dynamic
+
+    # make a path for the modcc build
+    mkdir build_modcc
+    cd build_modcc
+
+    # configure and make modcc
+    cmake ..
+    make -j modcc
+
+    # set PATH to find modcc
+    cd ..
+    export PATH=`pwd`/build_modcc/modcc:$PATH
```

Comment:
I use `-Duse_external_modcc=ON -DMODCC_BIN=$MODCC_BIN` instead. Maybe that option should go here as well?

Reply:
`use_external_modcc` is intended for this purpose (hence it is not an `option`).
I would rather leave it undocumented. Sam and I have some ideas for how to make cross compilation work "nearly always", just have to get around to implementing it when we have a bit more time.
God I hate CMake.